golem_sci/implementation.py

```diff
@@ -137,10 +137,15 @@ def __init__(self, geth_client, address, tx_sign=None, monitor=True):
         self._awaiting_transactions_lock = threading.Lock()
         self._awaiting_transactions = []
 
+        self._monitor_thread = None
```

Comment:
Why is it needed to have this member variable?

Reply:
It's not. But IMO it's better to store the reference.

Reply:
Why is it better? It's unused after all